go/vt/discovery/healthcheck.go

```diff
@@ -551,34 +451,45 @@ func (hc *HealthCheckImpl) checkConn(hcc *healthCheckConn, name string) {
 		}()
 
 		// Read stream health responses.
-		hcc.stream(streamCtx, hc, func(shr *querypb.StreamHealthResponse) error {
+		err := th.stream(streamCtx, func(shr *query.StreamHealthResponse) error {
 			// We received a message. Reset the back-off.
 			retryDelay = hc.retryDelay
 			// Don't block on send to avoid deadlocks.
 			select {
 			case servingStatus <- shr.Serving:
 			default:
 			}
-			return hcc.processResponse(hc, shr)
+			return th.processResponse(hc, shr)
 		})
 
 		// streamCancel to make sure the watcher goroutine terminates.
 		streamCancel()
 
+		if err != nil {
+			if strings.Contains(err.Error(), "health stats mismatch") {
+				hc.deleteConn(th.Tablet)
```

Comment:
I'm thinking `finalizeConn` should also `deleteConn`. Essentially, this goroutine and the existence of conn go hand-in-hand.

Reply:
deleteConn should actually be deleteTablet. I think the current name is confusing, I will rename it.